_overviews/overview/scala-tools.md

```diff
@@ -0,0 +1,320 @@
+
+In large projects, all of your Scala source code files will go under the *src/main/scala* and *src/test/scala* directories, but for a little sample project like this, you can put your source code file in the root directory of your project. Therefore, create a file named *HelloWorld.scala* in the root directory with these contents:
+
+```scala
+@main def helloWorld = println("Hello, world")
+```
+
+Now, use the `sbt run` command to compile and run your project. When you do so, you’ll see output that looks like this:
+
+```bash
+$ sbt run
+
+[info] welcome to sbt
+[info] loading settings for project ...
+[info] loading project definition
+[info] loading settings for project root from build.sbt ...
+[info] Compiling 1 Scala source ...
+[info] running helloWorld 
+Hello, world
+[success] Total time: 4 s
+```
+
+The first time you run `sbt` it downloads everything it needs, and that can take a little while to run, but after that it gets much faster. 
+
+Also, once you get this first step working, you’ll find that it’s much faster to run sbt interactively. To do that, first run the `sbt` command by itself:
+
+```bash
+$ sbt
+
+[info] welcome to sbt
+[info] loading settings for project ...
+[info] loading project definition ...
+[info] loading settings for project root from build.sbt ...
+[info] sbt server started at local:///Users/al/.sbt/1.0/server/7d26bae825c38a31074c/sock
+sbt:hello-world> _
+```
+
+Then inside this sbt shell, execute its `run` command:
+
+````
+sbt:hello-world> run
+
+[info] running helloWorld 
+Hello, world
+[success] Total time: 0 s
+````
+
+There, that’s much faster.
+
+If you type `help` at the sbt command prompt you’ll see a list of other commands you can run. But for now, just type `exit` to leave the sbt shell. (You can also press `CTRL-D` instead of typing `exit`.)
+
+
+### Other build tools for Scala
+
+<!-- TODO: add Fury? -->
+While sbt is widely used, there are other tools you can use to build Scala projects:
+
+- [Ant](https://ant.apache.org/)
+- [Gradle](https://gradle.org/)
+- [Maven](https://maven.apache.org/)
+- [Mill](https://www.lihaoyi.com/mill/)
+
+
+
+## Using sbt with ScalaTest
+
+ScalaTest is one of the main testing libraries for Scala projects, and in this section you’ll see how to create a Scala/sbt project that uses ScalaTest.
+
+
+### Creating the project directory structure
+
+As with the previous lesson, create an sbt project directory structure for a project named *HelloScalaTest* with the following commands:
+
+```sh
+mkdir HelloScalaTest
+cd HelloScalaTest
+mkdir -p src/{main,test}/{resources,scala}
+mkdir project target
+```
+
+
+### Creating the build.sbt file
+
+Next, create a *build.sbt* file in the root directory of your project with these contents:
+
+```scala
+name := "HelloScalaTest"
+version := "1.0"
+scalaVersion := "{{site.scala-version}}"
+
+libraryDependencies ++= Seq(
+  "org.scalatest" % "scalatest-core_0.27" % "3.2.2",
+  "org.scalatest" % "scalatest_0.27" % "3.2.2" % Test
```

Comment:
This should be updated to Scala 3.0

Reply:
I don’t think ScalaTest works with Scala 3 yet, though I could be wrong. “Dotty” [is mentioned on their website](https://www.scalatest.org/release_notes/3.1.0), but I don’t see any support listed [on Maven](https://mvnrepository.com/artifact/org.scalatest/scalatest).

Reply:
Digging a little further, I wondered where I got that `0.27` reference from, and I see [there may now be a ScalaTest version available](https://mvnrepository.com/artifact/org.scalatest/scalatest_3.0.0-M2) for 3.0.0-M2.

Reply:
This is updated.